pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -76,6 +76,26 @@ def __call__(self, *targs, **tkwargs):
             "The object on which the transform is applied is not valid. It can only be a tape, a QNode or a qfunc."
         )
 
+    def add_to_program(self, transform_program, *targs, **tkwargs):
```

Comment:
Wouldn't the transform program already have all the arguments and keyword arguments already bound?

How does this differ from the dunder method `__add__`?